homeassistant/components/zwave_js/select.py

```diff
@@ -173,7 +172,6 @@ def current_option(self) -> str | None:
 
     async def async_select_option(self, option: str) -> None:
         """Change the selected option."""
-        if (target_value := self._target_value) is None:
-            raise HomeAssistantError("Missing target value on device.")
+        assert self._target_value is not None
```

Comment:
If we know that this attribute is not None we should update the type annotation for the attribute so we don't need to assert in every method.

I don't see where in the discovery matching we require the target value property.

Reply:
you're right that we don't require it so theoretically a device could still have no target value. I suppose we can revert these changes and force a scenario like this to occur to ensure it is handled

Reply:
I don't know the z-wave spec details good enough to judge what's the best way. Can this case happen and when in that case?

Reply:
I don't believe it can. I think currentValue and targetValue are zwave-js concepts and for multilevel switches, which are the primary values for this type of cover, there should always be a way to set the switch to a new value

Reply:
Ok.